consensus/src/msg_handler.rs

```diff
@@ -8,41 +8,41 @@ use crate::commons::{ConsensusError, RoundUpdate};
 use crate::iteration_ctx::RoundCommittees;
 use crate::user::committee::Committee;
 use async_trait::async_trait;
-use node_data::ledger::to_str;
-use node_data::message::{Message, MessageTrait, Status};
+use node_data::message::{Message, Status};
 use node_data::StepName;
-use std::fmt::Debug;
 use tracing::{debug, trace};
 
 /// Indicates whether an output value is available for current step execution
 /// (Step is Ready) or needs to collect data (Step is Pending)
+#[allow(clippy::large_enum_variant)]
 pub enum HandleMsgOutput {
-    Pending(Message),
+    Pending,
     Ready(Message),
 }
 
 /// MsgHandler must be implemented by any step that needs to handle an external
 /// message within event_loop life-cycle.
 #[async_trait]
-pub trait MsgHandler<T: Debug + MessageTrait> {
+pub trait MsgHandler {
     /// is_valid checks a new message is valid in the first place.
     ///
     /// Only if the message has correct round and step and is signed by a
     /// committee member then we delegate it to Phase::verify.
     fn is_valid(
         &self,
-        msg: &T,
+        msg: &Message,
         ru: &RoundUpdate,
         iteration: u8,
         step: StepName,
         committee: &Committee,
         round_committees: &RoundCommittees,
     ) -> Result<(), ConsensusError> {
+        let from =
```

Comment:
I'd call this `signer` to avoid confusion with the sender of the message (our peer).
I'd probably rename also `get_pubkey_bls` to `get_signer`.
We can add these to the tech debt if you prefer

Reply:
Updated